app/data_model/answer.py

```diff
@@ -4,12 +4,13 @@
 
 
 class Answer:
-    def __init__(self, answer_id, value):
+    def __init__(self, answer_id, value, list_item_id=None):
         if answer_id is None or value is None:
             raise ValueError("Both 'answer_id' and 'value' must be set for Answer")
 
         self.answer_id = answer_id
         self.value = value
+        self.list_item_id = list_item_id
```

Comment:
When we had `group_instance_id` and it was giving a default of `None` to answers that were not 'repeats'.
I thought for the lists we wanted to only show list item id if it is a `ListCollector` and not default it to None?.

Reply:
What do you mean by 'show' list item id? I don't see the problem with having an optional `list_item_id` on the Answer, I don't think there's any other obvious way to do it?

Were you thinking of `group_instance` defaulting to 0 in the location?

Reply:
Show as in, assign a `list_item_id` key with a value only if it needs one. So I was referring to this:
``` json
{
      "answer_id": "repeating-anyone-else",
      "answer_instance": 0,
      "group_instance": 0,
      "group_instance_id": null,
      "value": "Yes"
},
```
Where `group_instance_id` was always assigning a value even if it didn't need one, I get that it's None. I just thought we said we wouldn't want to assign the key value if it wasn't needed. I don't really see a problem either way, it was just I thought that what we said.

Reply:
So, we can probably change the serialisation to ensure that list_item_id is not serialised if it is None. This would mean a custom json serialiser on here though I think?